scripts/cleanup_stale_session_audits.py

```
@@ -0,0 +1,618 @@
+        return {int(r.get("SessionId")): int(r.get("c") or 0) for r in rows}
+
+    def delete_audit_rows(self, table: str, session_ids: List[int]) -> int:
+        if not session_ids:
+            return 0
+        # D1 has a 100-bound-param cap per statement.
+        deleted = 0
+        for chunk_start in range(0, len(session_ids), 90):
+            chunk = session_ids[chunk_start: chunk_start + 90]
+            placeholders = ",".join("?" for _ in chunk)
+            cur = self._history.execute(
+                f"DELETE FROM {table} WHERE SessionId IN ({placeholders})",
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)